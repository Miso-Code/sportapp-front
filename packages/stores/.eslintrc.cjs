const eslintDefaultConfig = require('@sportapp/eslint-config/library')

module.exports = {
	...eslintDefaultConfig
}
