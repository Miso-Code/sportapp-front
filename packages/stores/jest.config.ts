export default {
	preset: 'ts-jest',
	setupFilesAfterEnv: ['<rootDir>/config/test/jest.setup.ts'],
	testEnvironment: 'jsdom',
	transform: {
		'^.+\\.tsx?$': ['ts-jest']
	},
	moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
	collectCoverage: true,
	collectCoverageFrom: [
		'<rootDir>/src/**/*.{ts,tsx}',
		'!<rootDir>/src/**/interfaces/**/*.{ts,tsx}',
		'!<rootDir>/src/**/__mocks__/**/*.{ts,tsx}'
	],
	moduleNameMapper: {
		'\\.(gif|ttf|eot|svg|png|jpg)$':
			'<rootDir>/config/test/__mocks__/fileMock.ts',
		'^.+\\.(css|less|scss)$': 'babel-jest',
		'^@/(.*)$': ['<rootDir>/src/$1']
	},
	coverageThreshold: {
		global: {
			branches: 70,
			functions: 70,
			lines: 70,
			statements: 70
		}
	},
	reporters: ['default']
}
