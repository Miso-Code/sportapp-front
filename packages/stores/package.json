{
	"name": "@sportapp/stores",
	"version": "1.0.0",
	"license": "MIT",
	"main": "src/index.ts",
	"type": "module",
	"scripts": {
		"format": "prettier --write \"**/*.{ts,tsx,md,js}\"",
		"lint": "yarn lint:js",
		"lint:js": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
		"pre-commit": "lint-staged",
		"test": "jest --ci --reporters=default --config jest.config.ts --no-cache --coverage --forceExit --detectOpenHandles --logHeapUsage --silent=false --verbose"
	},
	"dependencies": {
		"@react-native-async-storage/async-storage": "^1.23.1",
		"@sportapp/sportapp-repository": "*",
		"@tanstack/react-query": "^5.28.9",
		"@testing-library/jest-dom": "^6.4.2",
		"@testing-library/react": "^14.2.2",
		"@types/jest": "^29.5.12",
		"babel-jest": "^29.7.0",
		"jest": "^29.7.0",
		"prettier": "^3.2.5",
		"react": "^18.2.0",
		"ts-jest": "^29.1.2",
		"ts-jest-mock-import-meta": "^1.2.0",
		"ts-node": "^10.9.2",
		"zustand": "^4.5.2"
	},
	"devDependencies": {
		"@sportapp/eslint-config": "*",
		"@sportapp/prettier-config": "*",
		"@sportapp/typescript-config": "*",
		"eslint": "^8.19.0",
		"lint-staged": "^15.2.2",
		"simple-zustand-devtools": "^1.1.0"
	}
}
