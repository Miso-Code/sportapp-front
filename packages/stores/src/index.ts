export { useAuthStore, initialAuthState } from './auth'
export { useSportSessionStore, initialSportSessionState } from './sportSession'
export { useSportStore, initialSportState } from './sport'
export { useUserStore, initialUserState } from './user'
export {
	usePartnerAuthStore,
	initialAuthState as initialAuthStatePartner
} from './partner/auth'
export {
	useBusinessPartnerStore,
	initialBusinessPartnerState
} from './businessPartner'
export { useAlertStore, initialAlertState } from './alert'
export {
	initialProductPartnerState,
	usePartnerProductStore
} from './partner/products'

export { initialTrainingPlanState, useTrainingPlanStore } from './trainingPlan'

export { initialPaymentPlanState, usePaymentPlanStore } from './paymentPlan'

export { initialSportsmanState, useSportsmanStore } from './sportsman'

export { initialSportEventState, useSportEventStore } from './sportEvents'
