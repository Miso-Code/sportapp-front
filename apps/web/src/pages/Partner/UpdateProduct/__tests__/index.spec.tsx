import { usePartnerProductStore } from '@sportapp/stores'
import UpdateProductPartnerPage from '..'
import {
	fireEvent,
	render,
	RenderResult,
	waitFor
} from '@testing-library/react'

jest.mock('@sportapp/stores', () => ({
	usePartnerProductStore: jest.fn().mockReturnValue({
		createProduct: jest.fn(),
		error: undefined,
		loading: false,
		setError: jest.fn()
	}),
	usePartnerAuthStore: () => ({
		user: {
			id: '1',
			email: 'corre@correo.com',
			name: 'Corre',
			role: 'partner'
		}
	})
}))

jest.mock('@/utils/files', () => ({
	toBase64: jest.fn().mockResolvedValue('image64')
}))

describe('UpdateProductPartnerPage', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(<UpdateProductPartnerPage />)
	})

	afterEach(() => {
		wrapper.unmount()
	})

	it('should render the component', () => {
		expect(wrapper.container).toMatchSnapshot()
	})

	it('should call deleteProduct', () => {
		const deleteProduct = jest.fn()
		;(usePartnerProductStore as unknown as jest.Mock).mockReturnValue({
			createProduct: jest.fn(),
			error: undefined,
			loading: false,
			setError: jest.fn(),
			deleteProduct
		})
		wrapper.rerender(<UpdateProductPartnerPage />)
		const deleteButton = wrapper.getByText('productUpdate.delete')

		fireEvent.click(deleteButton as Element)

		waitFor(() => expect(deleteProduct).toHaveBeenCalled())
	})
})
