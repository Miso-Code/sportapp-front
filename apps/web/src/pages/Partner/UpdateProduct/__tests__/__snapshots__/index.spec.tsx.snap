// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateProductPartnerPage should render the component 1`] = `
<div>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorInherit MuiAppBar-positionStatic bg-white css-1cjrmu8-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-1lvtzne"
          >
            <button
              aria-controls="menu-appbar"
              aria-haspopup="true"
              aria-label="navbar.partner.menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root css-1t6c9ts"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1eyyvyt-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              title="navbar.partner.getProducts"
              type="button"
            >
              navbar.partner.getProducts
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1eyyvyt-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              title="navbar.partner.createProduct"
              type="button"
            >
              navbar.partner.createProduct
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1eyyvyt-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              title="navbar.partner.purchasedProducts"
              type="button"
            >
              navbar.partner.purchasedProducts
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root css-2uchni"
          >
            <button
              aria-label="navbar.open.profile"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  alt="Remy Sharp"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="/static/images/avatar/2.jpg"
                />
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <main
        class="my-8"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root p-8 max-w-4xl mx-auto css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 mb-4 css-5lbw0b-MuiTypography-root"
          >
            productUpdate.title
          </h4>
          <button
            class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-fullWidth mt-4 css-1pt4pzm-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            id=":r1:"
            tabindex="0"
            type="button"
          >
            productUpdate.delete
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </main>
    </div>
  </div>
  <div
    class="alert-partner-update-product-container MuiBox-root css-0"
  >
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-1i3m5il-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="ErrorOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              errors.partner.updateProduct.base
            </div>
            <div
              class="MuiAlert-action css-ki1hdl-MuiAlert-action"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
